proposals/0000-transient-storages.md

```diff
@@ -0,0 +1,336 @@
+
+This attempts to remove a content at a given key.
+
+- `ext_btree_storage_contains_item` with parameters:
+	- name : a pointer size to the name of a transient storage to rename.
+	- key : a pointer size to the key of the content to insert.
+Returns true when a key and content exists, false otherwhise.
+
+- `ext_btree_storage_get_item` with parameters:
+	- name : a pointer size to the name of a transient storage to rename.
+	- key : a pointer size to the key of the content to insert.
+Returns an optional bytes content containing the value associated with the given key.
+
+- `ext_btree_storage_read_item` with parameters:
+	- name : a pointer size to the name of a transient storage to rename.
+	- key : a pointer size to the key of the content to insert.
+	- `value_out` :  a pointer-size containing the buffer to which the value will be written to.
+	- `value_offset` : a u32 offset on the value.
+Returns an optional size of content written in the buffer (None if the content associated with key is not found).
+The size written can only differ from the buffer size if there is not enough content to read.
+
+- `ext_btree_storage_len_item` with parameters:
+	- name : a pointer size to the name of a transient storage to rename.
+	- key : a pointer size to the key of the content to insert.
+Returns an optional size of value content when the key exists.
+
+
+- `ext_btree_storage_count` with parameters:
+	- name : a pointer size to the name of a transient storage to rename.
+Returns an optinal u32 number of element currently in the btree.
+This should be a small cost operation (implementation needs to maintain a count of content).
+
+
+- `ext_btree_storage_hash32_item` with parameters:
+	- name : a pointer size to the name of a transient storage to rename.
+	- key : a pointer size to the key of the content to insert.
+	- algorithm: `Hash32Algorithm` passed as a byte, 0 for Blake2b256.
+Returns an optional 32 bytes buffer containing the hash of the value when the content exists.
+
+This can be a costy on big value, but then it would be the blob usecase, therefore no caching of result is expecting from implementation.
+
+- `ext_btree_storage_root32_item` with parameters:
+	- name : a pointer size to the name of a transient storage to rename.
+	- structure: `Root32Structure` passed as a byte, 0 for `SubstrateDefault` which is the same merkle trie implementation as the substrate trie.
+Returns an optional 32 bytes buffer containing the root hash for the current stae or nothing if there is no transient storage for the given name.
+
+This call is very costy. Implementation shall at least cache hash result when state did not change.
+
+- `ext_btree_storage_next_keys` with parameters:
+	- name : a pointer size to the name of a transient storage to rename.
+	- key: a pointer size to the previously accessed key.
+	- count: a u32 indicating the maximum number of key to return.
+Returns a scale encoded optional sized array of keys (rust `Option<Vec<Vec<u8>>>`).
```

Comment:
```suggestion
Returns a scale encoded optional sized array of keys (rust `Option<Vec<Vec<u8>>>`). Returns a SCALE-encoded `None` either if there is no transient btree with that name or if the `key` is the last key of that transient btree.
```

Reply:
Intention was more to:
Returns a SCALE-encoded `Some(vec![])` if the `key` is the last key of that transient storage.